### PR TITLE
Hide gas estimate on non-main network

### LINKED ISSUE #9106: Wrong gas estimate on non-main network
**Describe the bug**
When Metamask is connected with a custom network it seems to apply the gas estimates from Ethereum Mainnet.

**To Reproduce**
1. Connect Metamask with a custom network. I observed this with IDChain network, but probably this happens on all networks.
    RPC-URL: https://idchain.one/rpc/ 
    NetworkID: 74
1. Click on "Send" to send some eth
1. Select a destination address
1. Observe the gasprice proposed by Metamask: Right now it sets a value of 80gwei, which is way too high for IDChain (10gwei is the norm there)

**Expected behavior**
Metamask should not use ETH mainnet gas estimates for other networks.
If there is no gas price feed available it should force the user to set a price instead of chosing gas price from a different network.

**Browser details (please complete the following information):**
 - OS: Arch Linux
 - Browser: Brave Version 1.11.97 Chromium: 84.0.4147.89 (Offizieller Build) unknown (64-Bit)
 - MetaMask 8.0.5

### PR DESCRIPTION
**Pull request by PatrykLucka**
This PR removes advanced gas settings option and three-button gas input (slow, average, fast) and shows the gas fields instead on all non-main networks.

Fixes: #9106 

Screenshot 2020-08-12 at 16 49 00
Screenshot 2020-08-12 at 16 48 50

### CODE REVIEW

**Comment by rekmarks** [comment]
Hey Patryk 👋

To get the e2e tests passing, I _think_ we're going need to ensure that the advanced gas controls are enabled for non-Mainnet networks in tests, which can be done using the `process.env.IN_TEST` environment variable.

**Review by rekmarks — DISMISSED** [review]
LGTM!

**Comment by whymarrh** [comment]
Mind rebasing this on the latest `develop`? I think the `consistent-return` lint rule might want a value for the return.

File: ui/app/pages/send/send-content/send-gas-row/send-gas-row.component.js [hunk 1]
```diff
@@ -35,7 +35,11 @@ export default class SendGasRow extends Component {
 
   renderAdvancedOptionsButton () {
     const { metricsEvent } = this.context
-    const { showCustomizeGasModal } = this.props
+    const { showCustomizeGasModal, advancedInlineGasShown } = this.props
+    // Tests should behave in same way as mainnet, but are using Localhost
+    if (advancedInlineGasShown === true && !process.env.IN_TEST) {
```
Comment (rekmarks) [inline ui/app/pages/send/send-content/send-gas-row/send-gas-row.component.js:40]: nit: it'd be more consistent with everything to skip the `=== true`
```suggestion
    if (advancedInlineGasShown && !process.env.IN_TEST) {
```

**Review by rekmarks — DISMISSED** [review]

**Review by rekmarks — APPROVED** [review]

**Review by rekmarks — CHANGES_REQUESTED** [review]
@PatrykLucka upon review, I'm afraid I've failed to give you good advice on this PR. I'm sorry for the back-and-forth.

We should _not_ set the feature flag in order to implement this behavior. Rather, we should do this:
- If the network is Mainnet, the advanced gas controls showed be shown instead of the gas estimation _only if_ the `advancedInlineGas` feature flag is set to `true`.
- If the network is _not_ Mainnet, we should show the advanced gas controls, _regardless_ of the state of the `advancedInlineGas` feature flag.
  - _Unless_ we're in a test, in which case we should show the gas estimate.
    - We should only override the feature flag in tests if we have to. I think it's possible that we can leave it alone, and just make all networks behave like Mainnet in tests.

File: ui/app/store/actions.js [hunk 2]
```diff
@@ -1464,6 +1464,10 @@ export function setProviderType (type) {
     }
     dispatch(setPreviousProvider(currentProviderType))
     dispatch(updateProviderType(type))
+    // Tests should behave in same way as mainnet, but are using Localhost
+    if (!process.env.IN_TEST) {
+      dispatch(setFeatureFlag('advancedInlineGas', type !== 'mainnet'))
```
Comment (Gudahtt) [inline ui/app/store/actions.js:1469]: Wouldn't this have the effect of _enabling_ advanced inline gas for all users when they switched to mainnet? We'd still want this to be disabled unless the user turns it on.

Comment (Gudahtt) [reply ui/app/store/actions.js:1469]: :sweat_smile: @rekmarks this is covered by your comment

**Comment by PatrykLucka** [comment]
Makes sense, thanks!

File: ui/index.js [hunk 3]
```diff
@@ -111,7 +111,6 @@ async function startApp (metamaskState, backgroundConnection, opts) {
       id: unapprovedTxsAll[0].id,
     }))
   }
-
   backgroundConnection.on('update', function (metamaskState) {
```
Comment (rekmarks) [inline ui/index.js:114]: nit: unrelated whitespace deletion

**Review by rekmarks — DISMISSED** [review]
LGTM!

File: ui/app/components/app/gas-customization/gas-modal-page-container/gas-modal-page-container.container.js [hunk 4]
```diff
@@ -112,6 +110,8 @@ const mapStateToProps = (state, ownProps) => {
   const isMainnet = getIsMainnet(state)
   const showFiat = Boolean(isMainnet || showFiatInTestnets)
 
+  const hideBasic = state.appState.modal.modalState.props.hideBasic || (!isMainnet && !process.env.IN_TEST)
```
Comment (Gudahtt) [inline ui/app/components/app/gas-customization/gas-modal-page-container/gas-modal-page-container.container.js:113]: Unfortunately the gas chart shown on the 'Advanced' tab is _also_ built using information from Eth Gas Station, so hiding the basic tab isn't especially helpful. It's the 'Edit' button on the transaction confirmation screen that leads here that needs to be hidden:

advanced-gas-confirm

File: ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 5]
```diff
@@ -254,7 +256,7 @@ export default class ConfirmTransactionBase extends Component {
               onHeaderClick={() => this.handleEditGas()}
               secondaryText={hideFiatConversion ? this.context.t('noConversionRateAvailable') : ''}
             />
-            {advancedInlineGasShown
+            {advancedInlineGasShown || (!isMainnet && !process.env.IN_TEST)
```
Comment (Gudahtt) [inline ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js:259]: The `ConfirmDetailRow` component above is where the "Edit" button is that we need to hide.

It looks like the `ConfirmDetailRow` component expects the `headerText` prop to always be passed in. I'd suggest making that prop optional, and omitting all three related props (`headerText`, `headerTextClassName`, and `onHeaderClick`) when on testnets.

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Almost there! In my local testing, there is only one place remaining where ETH Gas Station data needs to be hidden.

**Review by Gudahtt — APPROVED** [review]
LGTM, thanks!

**Pull request merged**